core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/FileSearchQueryBuilder.scala

```diff
@@ -0,0 +1,95 @@
+package edu.uci.ics.texera.web.resource.dashboard
+import edu.uci.ics.texera.web.auth.SessionUser
+import edu.uci.ics.texera.web.model.jooq.generated.Tables.{FILE, USER, USER_FILE_ACCESS}
+import edu.uci.ics.texera.web.model.jooq.generated.enums.UserFileAccessPrivilege
+import edu.uci.ics.texera.web.model.jooq.generated.tables.pojos.File
+import edu.uci.ics.texera.web.resource.dashboard.DashboardResource.DashboardClickableFileEntry
+import edu.uci.ics.texera.web.resource.dashboard.FulltextSearchQueryUtils.{
+  getContainsFilter,
+  getDateFilter,
+  getFulltextSearchConditions
+}
+import edu.uci.ics.texera.web.resource.dashboard.user.file.UserFileResource.DashboardFile
+import org.jooq.{Condition, GroupField, OrderField, Record, TableLike}
+import org.jooq.impl.DSL
+
+import scala.jdk.CollectionConverters.CollectionHasAsScala
+object FileSearchQueryBuilder extends SearchQueryBuilder {
+
+  override val mappedResourceSchema: UnifiedResourceSchema = UnifiedResourceSchema(
+    resourceType = DSL.inline(SearchQueryBuilder.FILE_RESOURCE_TYPE),
+    name = FILE.NAME,
+    description = FILE.DESCRIPTION,
+    creationTime = FILE.UPLOAD_TIME,
+    fid = FILE.FID,
+    fileOwnerId = FILE.OWNER_UID,
+    fileUploadTime = FILE.UPLOAD_TIME,
+    filePath = FILE.PATH,
+    fileSize = FILE.SIZE,
+    userEmail = USER.EMAIL,
+    fileUserAccess = USER_FILE_ACCESS.PRIVILEGE
+  )
+
+  override protected def constructFromClause(
+      user: SessionUser,
+      params: DashboardResource.SearchQueryParams
+  ): TableLike[_] = {
+    FILE
+      .leftJoin(USER_FILE_ACCESS)
+      .on(USER_FILE_ACCESS.FID.eq(FILE.FID))
+      .leftJoin(USER)
+      .on(FILE.OWNER_UID.eq(USER.UID))
+      .where(USER_FILE_ACCESS.UID.eq(user.getUid))
+  }
+
+  override protected def constructWhereClause(
+      user: SessionUser,
+      params: DashboardResource.SearchQueryParams
+  ): Condition = {
+    val splitKeywords = params.keywords.asScala
+      .flatMap(_.split("[+\\-()<>~*@\"]"))
+      .filter(_.nonEmpty)
+      .toSeq
+
+    getFulltextSearchConditions(
+      splitKeywords,
+      List(FILE.NAME, FILE.DESCRIPTION)
+    )
+      .and(
+        getDateFilter(
+          params.creationStartDate,
+          params.creationEndDate,
+          FILE.UPLOAD_TIME
+        )
+      )
+      .and(getContainsFilter(params.owners, USER.EMAIL))
+  }
+
+  override protected def constructGroupByClause(
+      user: SessionUser,
```

Comment:
if they are helper functions instead of API endpoints, we don't need to pass in `user: SessionUser` right? I suppose those functions are user-independent.

Reply:
done, now we only pass uid.